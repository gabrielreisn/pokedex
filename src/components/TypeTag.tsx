import { TypeTagProps } from "./types";
import { pokemonTagType } from "../utils";

export const TypeTag = ({ value }: TypeTagProps) => (
  <span 
    style={{background: pokemonTagType(value), padding: '1px 3px', margin: '0 3px'}} 
    className='rounded'>
      {value}
  </span>
);