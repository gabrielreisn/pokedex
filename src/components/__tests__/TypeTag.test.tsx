import React from "react";
import { render } from "@testing-library/react";
import { TypeTag } from "../TypeTag";

const props = {
  value: "Normal"
};

describe("<TypeTag />", () => {
  it("matches the snapshot", () => {
    const wrapper = render(<TypeTag {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
