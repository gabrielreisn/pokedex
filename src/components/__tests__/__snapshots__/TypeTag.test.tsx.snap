// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TypeTag /> matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="rounded"
        style="background: rgb(168, 168, 120); padding: 1px 3px; margin: 0px 3px;"
      >
        Normal
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="rounded"
      style="background: rgb(168, 168, 120); padding: 1px 3px; margin: 0px 3px;"
    >
      Normal
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
